### PR TITLE
Fix #4028: Download platform tags and dirs

### LINKED ISSUE #4028: `nf-core pipeline downloads` issues when downloading pipelines for `--platform`
### Description of the bug

I am downloading several pipelines with  nf-core/tools version 3.5.2 and noticed two bugs in the process.

#### Initially missing git repository cache

On the first download attempt, the process fails when git is used to check out the pipeline revisions to run the container detection on it and edit the tags. But seemingly git is executed in the wrong directory. Due to the unique error message _Adapting your pipeline download unfortunately failed_, the exception can be precisely traced to

https://github.com/nf-core/tools/blob/fee44dae3fb1f86561f8ee11a62eb6207804dc5d/nf_core/pipelines/download/workflow_repo.py#L247-L250

In the example below, I am providing the output for a download of bamtofastq, but the issue applies to most other pipelines. 

Surprisingly, rerunning the identical command works without issues, indicating that now a cached version of the git repository in present in the correct place to execute the commands.

#### Outdir structure for `--platform` downloads with `-u 'copy'` / `-u 'remote'`

When downloading a pipeline as `--platform` download a bare git clone is created. If  container images are downloaded additionally, those folders are placed inside the bare clone:

```console
├── branches
├── hooks
├── info
├── objects
│   ├── info
│   └── pack
├── refs
│   ├── heads
│   └── tags
└── singularity-images
```

Instead, there should be a containing folder:


```console
├── pipeline.git
│   ├── branches
│   ├── hooks
│   ├── info
│   ├── objects
│   │   ├── info
│   │   └── pack
│   └── refs
│       ├── heads
│       └── tags
└── singularity-images
```


### Command used and terminal output

```console
nf-core pipelines download bamtofastq -r '2.2.0' --platform --outdir "bamtofastq.git" -s 'singularity' -u 'amend' -l 'quay.io' -l 'docker.io'


                                          ,--./,-.
          ___     __   __   __   ___     /,-._.--~\ 
    |\ | |__  __ /  ` /  \ |__) |__         }  {
    | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                          `._,._,'

    nf-core/tools version 3.5.2 - https://nf-co.re


WARNING  Could not find GitHub authentication token. Some API requests may fail.                                                                                                                       
INFO     Detected Nextflow version 25.10.3                                                                                                                                                             
INFO     Saving 'nf-core/bamtofastq'                                                                                                                                                                   
         Pipeline revision: '2.2.0'                                                                                                                                                                    
         Use containers: 'singularity'                                                                                                                                                                 
         Container library: 'quay.io, docker.io'                                                                                                                                                       
         Using NXF_SINGULARITY_CACHEDIR': /home/matthias/Pipelines/Download/singularity-cachedir'                                                                                                      
         Output file: 'bamtofastq.git.git'                                                                                                                                                             
         Enabled for Seqera Platform: 'True'                                                                                                                                                           
INFO     Downloading workflow for Seqera Platform                                                                                                                                                      
INFO     Collecting workflow from GitHub                                                                                                                                                               
ERROR    Adapting your pipeline download unfortunately failed:                                                                                                                                         
         Cmd('git') failed due to: exit code(1)                                                                                                                                                        
           cmdline: git checkout 2.2.0                                                                                                                                                                 
           stderr: 'error: pathspec '2.2.0' did not match any file(s) known to git'                                                                                                                    
                                                                                                                                                                                                       
ERROR    Cmd('git') failed due to: exit code(1)                                                                                                                                                        
           cmdline: git checkout 2.2.0                                                                                                                                                                 
           stderr: 'error: pathspec '2.2.0' did not match any file(s) known to git'                                                                                                                    


nf-core pipelines download bamtofastq -r '2.2.0' --platform --outdir "bamtofastq.git" -s 'singularity' -u 'amend' -l 'quay.io' -l 'docker.io'


                                          ,--./,-.
          ___     __   __   __   ___     /,-._.--~\ 
    |\ | |__  __ /  ` /  \ |__) |__         }  {
    | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                          `._,._,'

    nf-core/tools version 3.5.2 - https://nf-co.re


WARNING  Could not find GitHub authentication token. Some API requests may fail.                                                                                                                       
INFO     Detected Nextflow version 25.10.3                                                                                                                                                             
INFO     Saving 'nf-core/bamtofastq'                                                                                                                                                                   
         Pipeline revision: '2.2.0'                                                                                                                                                                    
         Use containers: 'singularity'                                                                                                                                                                 
         Container library: 'quay.io, docker.io'                                                                                                                                                       
         Using NXF_SINGULARITY_CACHEDIR': /home/matthias/Pipelines/Download/singularity-cachedir'                                                                                                      
         Output file: 'bamtofastq.git.git'                                                                                                                                                             
         Enabled for Seqera Platform: 'True'                                                                                                                                                           
INFO     Downloading workflow for Seqera Platform                                                                                                                                                      
INFO     Collecting workflow from GitHub                                                                                                                                                               
INFO     Fetching container names for workflow revision 2.2.0 using nextflow inspect. This might take a while.                                                                                         
INFO     Processing workflow revision 2.2.0, found 5 container images in total.
```

### System information


    Nextflow version: 25.10.3.10983
    OS: Linux Mint
    Version of nf-core/tools: 3.5.2
    Python version: 3.14

### PR DESCRIPTION
**Pull request by MatthiasZepper**
This pull request fixes the issues described in #4028. It 

- Corrects the `self.output_filename`
- Removes the parent from `bare_clone()` to ensure that the clone is actually inside the subdirectory named pipeline.git.
- Introduces an edge case handling in `tidy_tags_and_branches()` to avoid erroring out. 

Regarding the poor test coverage: I would like to defer this to #2940, when the whole logic must be reworked anyway. I don't really have time for it, but I can test if AI can actually help there. 

## PR checklist

- [x] This comment contains a description of changes (with reason)
- [x] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

**Comment by mashehu** [comment]
Just checking in as we are soon wrapping up for the release, how much is missing for this PR to be ready to be merged?

**Comment by MatthiasZepper** [comment]
> Just checking in as we are soon wrapping up for the release, how much is missing for this PR to be ready to be merged?

Essentially done. There is one issue that I identified during manual tests and that may be related or not. I still need to debug or at least identify the cause of that problem here:

```bash
nf-core pipelines download --platform -f -r "3.24.0" rnaseq --tag "3.24.0=rustqc" --outdir "foo" -s "singularity"


                                          ,--./,-.
          ___     __   __   __   ___     /,-._.--~\
    |\ | |__  __ /  ` /  \ |__) |__         }  {
    | \| |       \__, \__/ |  \ |___     \`-._,-`-,
                                          `._,._,'

    nf-core/tools version 3.6.0.dev0 - https://nf-co.re


WARNING  Could not find GitHub authentication token. Some API requests may fail.
INFO     Detected Nextflow version 25.10.4
WARNING  Deleting existing output directory: 'lalal'
INFO     Saving 'nf-core/rnaseq'
         Pipeline revision: '3.24.0'
         Use containers: 'singularity'
         Container library: 'quay.io'
         Using NXF_SINGULARITY_CACHEDIR': /Users/matthias.zepper/SingularityImages/nf-core-cache'
         Output file: 'foo/rnaseq.git'
         Enabled for Seqera Platform: 'True'
INFO     Downloading workflow for Seqera Platform
INFO     Collecting workflow from GitHub
INFO     Fetching container names for workflow revision 3.24.0 using nextflow inspect. This might take a while.
INFO     Processing workflow revision 3.24.0, found 46 container images in total.
Processing container images ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% • 0/90 tasks completed
Pulling singularity images ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━   0% • 0/44 tasks completed
ERROR    [Errno 2] No such file or directory: '/Users/matthias.zepper/SingularityImages/nf-core-cache/tmpuj46rgpn/tempfile' ->
         '/Users/matthias.zepper/SingularityImages/nf-core-cache/nvcr.io-nvidia-clara-clara-parabricks-4.6.0-1.img'

``` 
No idea why that doesn't work...

**Comment by MatthiasZepper** [comment]
Ready for review. That issue originated from a misconfiguration on my system and is not related to the PR.

File: nf_core/pipelines/download/workflow_repo.py [hunk 1]
```diff
@@ -294,11 +304,12 @@ def __add_additional_tags(self) -> None:
     def bare_clone(self, destination: Path):
         if self.repo:
             try:
-                destfolder = destination.parent.absolute()
+                destination = destination.absolute()
+                destfolder = destination.parent
                 if not destfolder.exists():
```
Comment (mashehu) [inline nf_core/pipelines/download/workflow_repo.py]:
```suggestion
```
not needed anymore if we use `mkdir()` with `exist_ok`. (couldn't fix the indentation of the next line in this suggestion unfortunately

**Review by mashehu — APPROVED** [review]
